File: src/main/java/xyz/emirdev/echogen/commands/PrefixCommand.java
package xyz.emirdev.echogen.commands;

import org.bukkit.entity.Player;

import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.tree.LiteralCommandNode;

import io.papermc.paper.command.brigadier.CommandSourceStack;
import io.papermc.paper.command.brigadier.Commands;
import xyz.emirdev.echogen.Echogen;
import xyz.emirdev.echogen.PluginCommand;
import xyz.emirdev.echogen.guis.PrefixGUI;

public class PrefixCommand extends PluginCommand {

    @Override
    public LiteralCommandNode<CommandSourceStack> getCommand() {
        return Commands.literal("prefix")
                .requires(ctx -> ctx.getSender().hasPermission("echogen.chat.prefix")
                        && Echogen.get().getPluginConfig().getRootNode().node("chat", "prefix", "enabled").getBoolean())
                .executes(this::execute)
                .build();
    }

    private int execute(CommandContext<CommandSourceStack> ctx) {
        Player player = getContextPlayer(ctx);
        if (player == null)
            return 1;

        PrefixGUI.openGUI(player);

        return 1;
    }
}

File: src/main/kotlin/cat/emir/echogen/commands/PrefixCommand.kt
package cat.emir.echogen.commands

import cat.emir.echogen.Echogen
import io.papermc.paper.command.brigadier.CommandSourceStack
import cat.emir.echolib.command.PluginCommand
import cat.emir.echolib.command.getPlayer
import cat.emir.echogen.guis.PrefixGUI
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.context.CommandContext

class PrefixCommand(plugin: Echogen) : PluginCommand<Echogen>(plugin) {

    override fun getCommand(): LiteralArgumentBuilder<CommandSourceStack> {
        return command("prefix") {
            requires {
                it.sender.hasPermission("echogen.chat.prefix") &&
                plugin.config.rootNode.node("chat", "prefix", "enabled").boolean
            }
            executes(::execute)
        }
    }

    fun execute(ctx: CommandContext<CommandSourceStack>): Int {
        val player = ctx.getPlayer() ?: return 1

        PrefixGUI(plugin, player).openGUI()

        return 1
    }
}
